doc/NextGen_ON_CONUS.md

```diff
@@ -0,0 +1,173 @@
+make -C extern/evapotranspiration/evapotranspiration/cmake_build && \
+cmake -B extern/sloth/cmake_build -S extern/sloth && \
+make -C extern/sloth/cmake_build && \
+cmake -B extern/SoilFreezeThaw/SoilFreezeThaw/cmake_build -S extern/SoilFreezeThaw/SoilFreezeThaw -DNGEN=ON && \
+cmake --build extern/SoilFreezeThaw/SoilFreezeThaw/cmake_build --target sftbmi -- -j 2 && \
+cmake -B extern/SoilMoistureProfiles/SoilMoistureProfiles/cmake_build -S extern/SoilMoistureProfiles/SoilMoistureProfiles -DNGEN=ON && \
+cmake --build extern/SoilMoistureProfiles/SoilMoistureProfiles/cmake_build --target smpbmi -- -j 2 &&
+```
+
+Copy the content into the file named `build_libs` and run the command:
+
+```
+source build_libs
+```
+
+This will build all libraries we need to run `ngen` at the time of this writing.
+
+Then, with the Python virtual environment activated, we can build the MPI executable using the following script:
+
+```
+cmake -S . -B cmake_build_mpi -DCMAKE_C_COMPILER=/local/lib/bin/mpicc -DCMAKE_CXX_COMPILER=/local/lib/bin/mpicxx \
+-DBOOST_ROOT=/home/shengting.cui/usr/boost_1_79_0/ \
+    -DNetCDF_ROOT=<path-to-NetCDF-ROOT-dir> \
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+    -DNGEN_IS_MAIN_PROJECT=ON \
+    -DNGEN_WITH_MPI:BOOL=ON                      \
+    -DNGEN_WITH_NETCDF:BOOL=ON                   \
+    -DNGEN_WITH_SQLITE:BOOL=ON                   \
+    -DNGEN_WITH_UDUNITS:BOOL=ON                  \
+    -DNGEN_WITH_BMI_FORTRAN:BOOL=ON              \
+    -DNGEN_WITH_BMI_C:BOOL=ON                    \
+    -DNGEN_WITH_PYTHON:BOOL=ON                   \
+    -DNGEN_WITH_ROUTING:BOOL=OFF                 \
+    -DNGEN_WITH_TESTS:BOOL=ON                    \
+    -DNGEN_QUIET:BOOL=ON                         \
+    -DNGEN_WITH_EXTERN_SLOTH:BOOL=ON             \
+    -DNGEN_WITH_EXTERN_TOPMODEL:BOOL=OFF         \
+    -DNGEN_WITH_EXTERN_CFE:BOOL=OFF              \
+    -DNGEN_WITH_EXTERN_PET:BOOL=OFF              \
+    -DNGEN_WITH_EXTERN_NOAH_OWP_MODULAR:BOOL=ON
+cmake --build cmake_build_mpi --target all -j 8
+```
+
+For the meaning of each option in the script, see `ngen/wiki` [build](https://github.com/NOAA-OWP/ngen/wiki/Building) page.
+
+Suppose the above script is named `build_mpi`, execute the following command to build:
+
+`source build_mpi`
+
+This will build an executable in the `cmake_build_mpi` directory named `ngen` and another named `partitionGenerator` as well as all the unit tests in the `cmake_build_mpi/test`.
+
+# CONUS Hydrofabric
+
+The CONUS hydrofabric is downloaded from [here](https://www.lynker-spatial.com/#v20.1/). The file name under the list is `conus.gpkg`. It is cautioned that since the data there are evolving and newer version may be available in the future. When using a newer version, be mindful that the corresponding initial configuration file generation and validation for all submodules at CONUS scale are necessary, which may be a non-trivial process due to the shear size of the spatial scale.
```

Comment:
```suggestion
The CONUS hydrofabric is downloaded from [here](https://www.lynker-spatial.com/#v20.1/). The file name under the list is `conus.gpkg`. Note that since the data there is continually evolving, a newer version may be available in the future. When using a newer version, be mindful that the corresponding initial configuration file generation and validation for all submodules at CONUS scale is necessary, which may be a non-trivial process due to the sheer size of the spatial scale.
```